x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -750,9 +751,10 @@ export class MapEmbeddable
     }
 
     if (areLayersLoaded(this._savedMap.getStore().getState())) {
-      const layers = getLayerList(this._savedMap.getStore().getState());
+      const mapState = this._savedMap.getStore().getState();
+      const layers = getLayerList(mapState);
       const isLoading =
-        layers.length === 0 ||
+        getWaitingForMapReadyLayerListRaw(mapState).length > 0 ||
```

Comment:
Thinking about this more, instead of getWaitingForMapReadyLayerListRaw, you can just use `getMapReady`. I think that is a little more clear and does not leak as much implemenation detail into the embeddable.

On additional change to make would be to update [mapReady](https://github.com/elastic/kibana/blob/main/x-pack/plugins/maps/public/actions/map_actions.ts#L162) to move `dispatch({type: MAP_READY})` to end of method to avoid any timing issues where mapReady set but before layerList state is updated